### ISSUE TITLE
g++ linker errors

### ISSUE THREAD

**Issue by shsr04** [thm-mni-ii/sea#46]
(`visual` branch) When using g++, the build fails due to a linker error: https://travis-ci.org/thm-mni-ii/sea/builds/437500592

```
[100%] Linking CXX executable bin/main
CMakeFiles/main.dir/src-view/examples.cpp.o: In function `SealibVisual::VisualDFS::run()':
examples.cpp:(.text+0x206): undefined reference to `void Sealib::DFS::process_small<Sealib::ExtendedSegmentStack>(unsigned int, Sealib::Graph*, Sealib::CompactArray*, Sealib::ExtendedSegmentStack*, void (*)(unsigned int, Sealib::Graph*, Sealib::CompactArray*, Sealib::ExtendedSegmentStack*), std::function<void (unsigned int)>, std::function<void (unsigned int, unsigned int)>, std::function<void (unsigned int, unsigned int)>, std::function<void (unsigned int)>)'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/main] Error 1
make[1]: *** [CMakeFiles/main.dir/all] Error 2
make: *** [all] Error 2
```

---

**Comment by andrej-sajenko** [comment]
The build still fails when using g++ compiler.

I suggest we just start to fix all the code smells. Then the g++ compiler should run as well.
* https://sonarcloud.io/project/issues?id=thm-mni-ii_sea&resolved=false&types=CODE_SMELL

**Comment by shsr04** [comment]
Fixed with #87
